### ISSUE TITLE
Can't access charge in CP

### ISSUE THREAD

**Issue by twd3** [silentzco/statamic-charge#20]
Updated to 4.0.5 and tested frontend. Everything went perfect. Logged in to CP to make sure everything showed up right in there and got: "Trying to get property 'email' of non-objectC:\_statamic\ma\storage\framework\views\c01a9dcf54fa877923f7e925c6a3fd949db9e2cc.php#21" and that new test subscription has an email (I can see it in stripe).
```
Trying to get property 'email' of non-object (View: C:\_statamic\ma\vendor\silentz\charge\resources\views\cp\subscriptions.blade.php)C:\_statamic\ma\vendor\silentz\charge\resources/views/cp/subscriptions.blade.php#23Facade\Ignition\Exceptions\ViewException
    Trying to get property 'email' of non-objectC:\_statamic\ma\storage\framework\views\c01a9dcf54fa877923f7e925c6a3fd949db9e2cc.php#21ErrorException

                <tbody>
                    <?php $__currentLoopData = $subscriptions; $__env->addLoop($__currentLoopData); foreach($__currentLoopData as $subscription): $__env->incrementLoopIndices(); $loop = $__env->getLastLoop(); ?>
                        <tr>
                            <td><?php echo e($subscription->user->email); ?></td>
                            <td><?php echo e($subscription->stripe_plan); ?></td>
                            <td><?php echo e($subscription->stripe_status); ?></td>
                            <td><?php echo e($subscription->ends_at); ?></td>

    #0 C:\_statamic\ma\storage\framework\views\c01a9dcf54fa877923f7e925c6a3fd949db9e2cc.php(21): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8, 'Trying to get p...', 'C:\\_statamic\\ma...', 21, Array)
    #1 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\View\Engines\PhpEngine.php(41): include('C:\\_statamic\\ma...')
    #2 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\View\Engines\CompilerEngine.php(57): Illuminate\View\Engines\PhpEngine->evaluatePath('C:\\_statamic\\ma...', Array)
    #3 C:\_statamic\ma\vendor\facade\ignition\src\Views\Engines\CompilerEngine.php(37): Illuminate\View\Engines\CompilerEngine->get('C:\\_statamic\\ma...', Array)
    #4 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\View\View.php(139): Facade\Ignition\Views\Engines\CompilerEngine->get('C:\\_statamic\\ma...', Array)
    #5 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\View\View.php(122): Illuminate\View\View->getContents()
    #6 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\View\View.php(91): Illuminate\View\View->renderContents()
    #7 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Http\Response.php(62): Illuminate\View\View->render()
    #8 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Http\Response.php(34): Illuminate\Http\Response->setContent(Object(Illuminate\View\View))
    #9 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Router.php(759): Illuminate\Http\Response->__construct(Object(Illuminate\View\View), 200, Array)
    #10 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Router.php(731): Illuminate\Routing\Router::toResponse(Object(Illuminate\Http\Request), Object(Illuminate\View\View))
    #11 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Router.php(685): Illuminate\Routing\Router->prepareResponse(Object(Illuminate\Http\Request), Object(Illuminate\View\View))
    #12 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(128): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
    #13 C:\_statamic\ma\vendor\statamic\cms\src\Http\Middleware\CP\CountUsers.php(18): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #14 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Statamic\Http\Middleware\CP\CountUsers->handle(Object(Illuminate\Http\Request), Object(Closure))
    #15 C:\_statamic\ma\vendor\statamic\cms\src\Http\Middleware\CP\Localize.php(16): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #16 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Statamic\Http\Middleware\CP\Localize->handle(Object(Illuminate\Http\Request), Object(Closure))
    #17 C:\_statamic\ma\vendor\statamic\cms\src\Http\Middleware\CP\Authorize.php(25): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #18 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Statamic\Http\Middleware\CP\Authorize->handle(Object(Illuminate\Http\Request), Object(Closure))
    #19 C:\_statamic\ma\vendor\statamic\cms\src\Http\Middleware\CP\ContactOutpost.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #20 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Statamic\Http\Middleware\CP\ContactOutpost->handle(Object(Illuminate\Http\Request), Object(Closure))
    #21 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Middleware\SubstituteBindings.php(41): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #22 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
    #23 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\VerifyCsrfToken.php(77): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #24 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
    #25 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\View\Middleware\ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #26 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
    #27 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Session\Middleware\StartSession.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #28 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Session\Middleware\StartSession.php(62): Illuminate\Session\Middleware\StartSession->handleStatefulRequest(Object(Illuminate\Http\Request), Object(Illuminate\Session\Store), Object(Closure))
    #29 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
    #30 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #31 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
    #32 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Cookie\Middleware\EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #33 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
    #34 C:\_statamic\ma\vendor\statamic\cms\src\Http\Middleware\SwapExceptionHandler.php(19): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #35 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Statamic\Http\Middleware\SwapExceptionHandler->handle(Object(Illuminate\Http\Request), Object(Closure))
    #36 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #37 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Router.php(687): Illuminate\Pipeline\Pipeline->then(Object(Closure))
    #38 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Router.php(662): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
    #39 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Router.php(628): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
    #40 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Routing\Router.php(617): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
    #41 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(165): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
    #42 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(128): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
    #43 C:\_statamic\ma\vendor\statamic\cms\src\Http\Middleware\CheckMultisite.php(14): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #44 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Statamic\Http\Middleware\CheckMultisite->handle(Object(Illuminate\Http\Request), Object(Closure))
    #45 C:\_statamic\ma\vendor\statamic\cms\src\Http\Middleware\PoweredByHeader.php(19): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #46 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Statamic\Http\Middleware\PoweredByHeader->handle(Object(Illuminate\Http\Request), Object(Closure))
    #47 C:\_statamic\ma\vendor\barryvdh\laravel-debugbar\src\Middleware\InjectDebugbar.php(65): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #48 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Barryvdh\Debugbar\Middleware\InjectDebugbar->handle(Object(Illuminate\Http\Request), Object(Closure))
    #49 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #50 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
    #51 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #52 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
    #53 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #54 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
    #55 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode.php(63): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #56 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
    #57 C:\_statamic\ma\vendor\fruitcake\laravel-cors\src\HandleCors.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #58 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Fruitcake\Cors\HandleCors->handle(Object(Illuminate\Http\Request), Object(Closure))
    #59 C:\_statamic\ma\vendor\fideloper\proxy\src\TrustProxies.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #60 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(167): Fideloper\Proxy\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
    #61 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
    #62 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(140): Illuminate\Pipeline\Pipeline->then(Object(Closure))
    #63 C:\_statamic\ma\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php(109): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
    #64 C:\_statamic\ma\public\index.php(55): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
    #65 C:\Users\Duke\AppData\Roaming\Composer\vendor\cretueusebiu\valet-windows\server.php(143): require('C:\\_statamic\\ma...')
    #66 {main}
```

**Comment by edalzell** [comment]
You have a subscription without a user? How can that be?

Can you check your db to see if everything is all there? If you look in the `subscriptions` table you should see rows with a `user_id`, then the matching user in the `users` table.

**Comment by twd3** [comment]
Yeah, the new user amde it to the database and the user has a stripe id and the users id is in the subscriptions table.

**Comment by edalzell** [comment]
do all your subscriptions have users?

**Comment by twd3** [comment]
Some of the users I see here in the subscriptions table have been deleted.

**Comment by twd3** [comment]
Yup, as soon as I deleted the rows of deleted users in the subscriptions table charge in cp started working.

**Comment by edalzell** [comment]
How did you delete the users @twd3 ?

**Comment by twd3** [comment]
In the cp, I first deleted test subscriptions from stripe then deleted users in cp.

**Comment by edalzell** [comment]
How did you "delete" the subscription? Do you mean cancel?

**Comment by twd3** [comment]
Yeah canceled at stripe I guess

**Comment by edalzell** [comment]
Right, that's not deleting it, only cancelling it.

No worries, I made the code a bit more resilient. Will be in the next release.

**Comment by edalzell** [comment]
fixed in 4.0.6
